### PR TITLE
[tests-only] [full-ci] adds function param and return type declaration

### PR DESCRIPTION
**Pull request by sakshamgurung**
## Description

- Adds function parameters and return type declaration for existing context files
- Added for:
- [x] CalDavContext.php
- [x] CardDavContext.php
- [x] WebUIGeneralContext.php
- [x] WebUIFilesContext.php
## Related Issue

- Working on issue: https://github.com/owncloud/core/issues/39038
- Complete lists: https://github.com/owncloud/core/issues/39243

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require increase of minor version instead of patch)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 
- [ ] Changelog item, see [TEMPLATE](https://github.com/owncloud/core/blob/master/changelog/TEMPLATE)

### CODE REVIEW

File: tests/acceptance/features/bootstrap/WebUIGeneralContext.php [hunk 1]
```diff
@@ -167,15 +167,15 @@ public function __construct(
...
 	}
 
 	/**
 	 *
 	 * @return OwncloudPage
 	 */
-	public function getCurrentPageObject() {
+	public function getCurrentPageObject():OwncloudPage {
 		return $this->currentPageObject;
 	}
```
Comment (sakshamgurung) [inline tests/acceptance/features/bootstrap/WebUIGeneralContext.php:181]: Giving return type ```string``` in ```public function getCurrentServer():string {...}``` and running test that uses this function gives error: ```Return value of WebUIGeneralContext::getCurrentServer() must be of the type string, null returned```

**Comment by sakshamgurung** [comment]
Some common error after declaring param and return type of function in WebUIGeneralContext.php
- ```When the user reloads the current page of the webUI```
```Type error: Return value of WebUIGeneralContext::getCurrentPageObject() must be an instance of Page\OwncloudPage, null returned (Behat\Testwork\Call\Exception\FatalThrowableError)```
- ```Given the setting "([^"]*)" in the section "([^"]*)" has been (disabled/enabled)```
```Type error: Return value of WebUIGeneralContext::getSavedCapabilitiesXml() must be of the type string, array returned (Behat\Testwork\Call\Exception\FatalThrowableError)```

I will try to solve it, but if anyone has an idea then reply below.

**Comment by kiranparajuli589** [comment]
is it possible to set return type like (string|null) 🤔 ?? that may solve your problem

**Comment by saw-jan** [comment]
> is it possible to set return type like (string|null)  ?? that may solve your problem

yeah, I think it is possible with PHP 8

**Comment by sakshamgurung** [comment]
> > is it possible to set return type like (string|null)  ?? that may solve your problem
> 
> yeah, I think it is possible with PHP 8

It is not possible below PHP 8

**Comment by phil-davis** [comment]
> Are we using PHP 8 in drone and project?

oC10 only supports up to PHP 7.4. The Behat test runner will work with PHP 8.0, so we could run the acceptance test step with PHP 8.0 and the server-under-test with PHP 7.4. But that will be a real pain for testing locally - you would have to sort out how to run the oC10 server with PHP 7.4 and switch to using PHP 8.0 in the terminal window where you run Behat.

For now, it is best to use only PHP 7.4 features. Usually we are happy to run everything locally on PHP 7.4. PHP 7.2 is about to be dropped, and if we want to test the oC10 server on PHP 7.3, then we can run it in CI.

File: tests/acceptance/features/bootstrap/WebUIGeneralContext.php [hunk 2]
```diff
@@ -642,9 +642,12 @@ public function theUserReloadsTheCurrentPageOfTheWebUI() {
 	/**
 	 * returns the saved capabilities as XML
 	 *
-	 * @return string
+	 * @return array
 	 */
-	public function getSavedCapabilitiesXml() {
+	public function getSavedCapabilitiesXml():array {
+		echo "Hello there: ";
+		print_r($this->savedCapabilitiesXml);
```
Comment (sakshamgurung) [inline tests/acceptance/features/bootstrap/WebUIGeneralContext.php:649]: I will remove this in next push

File: tests/acceptance/features/bootstrap/WebUIFilesContext.php [hunk 3]
```diff
@@ -2269,9 +2274,9 @@ public function theUserOpensTheFileActionMenuOfFileFolderOnTheWebui($name) {
 	 *
 	 * @param string $fileName Name of the file/Folder
 	 *
-	 * @return ResponseInterface
+	 * @return void
```
Comment (sakshamgurung) [inline tests/acceptance/features/bootstrap/WebUIFilesContext.php:2277]: There was no return from this function. So, changed it to `void`. If it is suppose to return any thing please inform.

**Comment by phil-davis** [comment]
https://drone.owncloud.com/owncloud/core/32045/3/6
Fails code style.

File: tests/acceptance/features/bootstrap/WebUIFilesContext.php (hunk 3 above)
Comment (phil-davis) [reply tests/acceptance/features/bootstrap/WebUIFilesContext.php:2277]: That is correct - there is no return value. The response is saved in the context object(s).

**Review by phil-davis — APPROVED** [review]
LGTM if CI passes.

**Comment by phil-davis** [comment]
https://drone.owncloud.com/owncloud/core/32050/154/15
```
Type error: Argument 1 passed to WebUIFilesContext::theUserBrowsesDirectlyToDetailsTabOfFileInFolder() must be of the type string, null given, called in /drone/src/tests/acceptance/features/bootstrap/WebUIFilesContext.php on line 361 (Behat\Testwork\Call\Exception\FatalThrowableError)
```

**Pull request merged**